src/apps/the-exchange/hooks/useOffer.tsx

```diff
@@ -1,60 +1,85 @@
-import { BridgingProvider, ExchangeOffer } from '@etherspot/prime-sdk/dist/sdk/data';
+import { ExchangeOffer } from '@etherspot/prime-sdk/dist/sdk/data';
 import { useEtherspotSwaps } from '@etherspot/transaction-kit';
 import { parseUnits } from 'ethers/lib/utils';
 import { processEth } from '../utils/blockchain';
+import { Route } from '@lifi/sdk';
 
-type SwapOffer = {
-    fromAmount: number,
-    fromTokenAddress: string,
-    fromChainId: number;
-    fromTokenDecimals: number;
-    toTokenAddress: string,
-    toChainId: number,
-    toTokenDecimals: number;
-    fromAccountAddress?: string
-    slippage?: number;
-    provider?: BridgingProvider
-    
-}
+// types
+import { SwapOffer, SwapType } from '../utils/types';
 
 const useOffer = (chainId: number) => {
     const { getOffers } = useEtherspotSwaps(chainId);
 
-        const getBestOffer = async ({ fromAmount, fromTokenAddress, fromChainId, fromTokenDecimals, toTokenAddress, toChainId, toTokenDecimals, fromAccountAddress, slippage, provider }: SwapOffer): Promise<ExchangeOffer | undefined> => {
+        const getBestOffer = async ({ fromAmount, fromTokenAddress, fromChainId, fromTokenDecimals, toTokenAddress, toChainId, toTokenDecimals }: SwapType): Promise<SwapOffer | undefined> => {
+            let selectedOffer: SwapOffer | undefined = undefined;
+
+            // uses getAdbvanceRoutesLifi (Lifi) - different chains, different tokens
             if (fromChainId !== toChainId) {
-                return;
-            }
+                try {
+                    const allOffersResponse = await getOffers(parseUnits(`${fromAmount}`, fromTokenDecimals), fromTokenAddress, toTokenAddress, toChainId);
+                    
+                    const allOffers = allOffersResponse?.offers as Route[];
 
-            // TO DO - remove the if statement below when other functions are added to get offers
-            if (fromAccountAddress || slippage || provider) {
-                return;
+                    if (allOffers.length) {
+                        const bestOffer = allOffers.reduce((a,b) => {
+                            const receiveAmountA = processEth(a.toAmount, toTokenDecimals);
+                            const receiveAmountB = processEth(b.toAmount, toTokenDecimals)
+                            return (
+                                receiveAmountA > receiveAmountB ? a : b
+                            )
+                        })
+    
+                        selectedOffer = {
+                            tokenAmountToReceive: processEth(bestOffer.toAmount, toTokenDecimals),
+                            offer: bestOffer as Route,
+                        };
+                    }
+                    
+                } catch (e) {
+                    console.error('Sorry, an error occured while trying to fetch the best swap offer. Please try again.', e);
```

Comment:
So this looks like if there is an error, `undefined` is returned back to the user. This should ideally return an error here to allow another developer or the UI layer to "catch" this and react (mind the pun) to it